### PR TITLE
UI: arise bot

### PR DESCRIPTION
**Pull request by eranra**
This is a bot that will allow the users to use NL to complete the configuration and work with the UI. 
It is using LLM service from HuggingFace based on the `hugchat` python lib. Free and using a strong "CohereForAI/c4ai-command-r-plus-08-2024" LLM model :-) 

In this PR the bot acts as an advisor and doesn't actuate to change the configuration UI. In future versions of the bot, such capabilities might be added

image

A new secret file needs to be created for the bot to work. Without that file, that bot is not activated
The file needs to be located under the `ui` folder. under folder names `.streamlit` and called `secrets.toml`. The content is::

```
eranra@eranra-wsl [8204] ~/go/src/github.com/arise-insights/arise-predictions/ui
$ cat .streamlit/secrets.toml
BOT_EMAIL = "arise.predictions@gmail.com"
BOT_PASSWORD = "************"
```

The same content of the file is set under the streamlit cloud secret section. Exactly cut and paste the file :-)

### CODE REVIEW

File: ui/README.md [hunk 1]
```diff
@@ -19,4 +20,8 @@ streamlit run main.py
 
 Change the config.yaml file to fit your custom configuration.  
 The provided version from github is aligned with the demo mlcommon
```
Comment (rachelt44) [inline ui/README.md:22]: @eranra link to the mlcommons example folder in the repo

File: ui/docs/how_to_use_the_ui.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  Displays prediction results and supports additional interactions.
+
+- **Interactive Query Panel:**  
+  Found at the bottom right, this AI Bot lets users ask questions or explore additional features.
+
+---
+
+## Configuring Input Parameters
+
+Input parameters are set in the **Prediction Configuration Panel** and are essential for accurate results. The available fields depend on the dataset and model used. Below are the configurable input parameters for the `MLCommons` dataset:
```
Comment (rachelt44) [inline ui/docs/how_to_use_the_ui.md:38]: @eranra link to the mlcommons example folder in the repo

File: ui/docs/how_to_use_the_ui.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+7. **Scenario:**  
+   - **Purpose:** Define the workload scenario.  
+   - **Configuration:** Select options like *Offline* or *Server*.
+
+---
+
+## Configuring Output Parameters
+
+The **Output Configuration Panel** allows users to select prediction metrics. For the `MLCommons` dataset, available outputs include:
```
Comment (rachelt44) [inline ui/docs/how_to_use_the_ui.md:72]: @eranra link to the mlcommons example folder in the repo

File: ui/docs/how_to_use_the_ui.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+   - **Purpose:** Display the model's token generation rate based on the configuration.  
+   - **Configuration:** Check the box next to this metric.
+
+> **Note:** Multiple outputs can be selected for a comprehensive analysis.
+
+---
+
+## Interpreting the Results Table
+
+After configuring inputs and outputs, click the **Predict** button. Results appear in the **Main Display Area**. For the `MLCommons` dataset, the table includes:
```
Comment (rachelt44) [inline ui/docs/how_to_use_the_ui.md:84]: @eranra link to the mlcommons example folder in the repo

File: ui/docs/how_to_use_the_ui.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+### How to Use:
+1. Click the **Add Filters** button to open a filter selection combo box.
+2. Select the column to filter.
+3. Use the dropdown or slider to narrow the data focus.
+
+---
+
+## Example Use Case
+
+**Scenario:** Estimate the resources required to achieve 5000 tokens per second for an AI model using NVIDIA H100 GPUs.
```
Comment (rachelt44) [inline ui/docs/how_to_use_the_ui.md:120]: Add a comment that a more elaborate example and explanation of what happens under the hood can be found at ../doc/example.md

**Review by rachelt44 — CHANGES_REQUESTED** [review]
Looks good. Put some inline comments on the UI documentation.

File: ui/docs/how_to_use_the_ui.md (hunk 2 above)
Comment (eranra) [reply ui/docs/how_to_use_the_ui.md]: fixed

File: ui/docs/how_to_use_the_ui.md (hunk 3 above)
Comment (eranra) [reply ui/docs/how_to_use_the_ui.md]: fixed

File: ui/docs/how_to_use_the_ui.md (hunk 4 above)
Comment (eranra) [reply ui/docs/how_to_use_the_ui.md:87]: fixed

File: ui/docs/how_to_use_the_ui.md (hunk 5 above)
Comment (eranra) [reply ui/docs/how_to_use_the_ui.md:123]: fixed

File: ui/README.md (hunk 1 above)
Comment (eranra) [reply ui/README.md]: fixed

**Review by rachelt44 — APPROVED** [review]

**Pull request merged**